web_m2x_options/static/src/js/form.js

```diff
@@ -96,272 +107,255 @@ odoo.define("web_m2x_options.web_m2x_options", function (require) {
+                ),
+                m2x_create_edit = is_option_set(
+                    ir_options["web_m2x_options.create_edit"]
+                );
+
+            if (!create_edit_undef) {
+                return create_edit;
+            }
+
+            if (!m2x_create_edit_undef) {
+                return m2x_create_edit;
+            }
+
+            return true;
+        },
+
+        _canFilterDialogRemoveButtons: function () {
+            var m2o_dialog_remove_buttons_undef = _.isUndefined(
+                    ir_options["web_m2x_options.m2o_dialog_remove_buttons"]
+                ),
+                m2o_dialog_remove_buttons = is_option_set(
+                    ir_options["web_m2x_options.m2o_dialog_remove_buttons"]
+                );
+            if (!m2o_dialog_remove_buttons_undef) {
+                return m2o_dialog_remove_buttons;
+            }
+            return false;
+        },
+
+        _canShowDialogQuickCreate: function () {
+            return this._canShowQuickCreate() || !this._canFilterDialogRemoveButtons();
+        },
+
+        _canShowDialogCreateAndEdit: function () {
+            return (
+                this._canShowCreateAndEdit() || !this._canFilterDialogRemoveButtons()
+            );
+        },
+
+        /**
+         * @private
+         * @param {OdooEvent} event
+         * @returns {Array}
+         */
+        _onSearchCreatePopup: function (event) {
+            const valueContext = this._createContext(event.data.value);
+            this.el.querySelector(":scope input").value = "";
+            return this._searchCreatePopup("form", false, valueContext);
+        },
+
         _onInputFocusout: function () {
             var m2o_dialog_opt =
                 is_option_set(this.nodeOptions.m2o_dialog) ||
                 (_.isUndefined(this.nodeOptions.m2o_dialog) &&
                     is_option_set(ir_options["web_m2x_options.m2o_dialog"])) ||
                 (_.isUndefined(this.nodeOptions.m2o_dialog) &&
                     _.isUndefined(ir_options["web_m2x_options.m2o_dialog"]));
-            if (this.can_create && this.floating && m2o_dialog_opt) {
+            if (this.floating && m2o_dialog_opt) {
+                if (
+                    !this._canCreate() ||
+                    (!this._canShowDialogQuickCreate() &&
+                        !this._canShowDialogCreateAndEdit())
+                ) {
+                    // If we cannot carry out either create or create and edit
+                    // actions then we should not show the dialog and we should zero
+                    // out the input
+                    this.el.querySelector(":scope input").value = "";
+                    return;
+                }
+
                 new M2ODialog(this, this.string, this.$input.val()).open();
             }
         },
 
-        _search: function (search_val) {
-            var self = this;
+        _search: async function (searchValue = "") {
```

Comment:
From my tests, this kind of statement doesn't work very well with inheritance. I'm not sure if it's still happening. I would prefer a more classic use of promises (new Promise...).